lib/inspector.js

```diff
@@ -0,0 +1,70 @@
+'use strict';
+
+const connect = process.binding('inspector').connect;
+const EventEmitter = require('events');
+
+if (!connect)
+  throw new Error('Inspector is not available');
+
+const connectionSymbol = Symbol('connectionProperty');
+const messageCallbacksSymbol = Symbol('messageCallbacks');
+const nextIdSymbol = Symbol('nextId');
+const onMessageSymbol = Symbol('onMessage');
+
+class Session extends EventEmitter {
+  constructor() {
+    super();
+    this[connectionSymbol] =
+        connect((message) => this[onMessageSymbol](message));
+    this[nextIdSymbol] = 1;
+    this[messageCallbacksSymbol] = new Map();
+  }
+
+  [onMessageSymbol](message) {
+    const parsed = JSON.parse(message);
+    if (parsed['id']) {
+      const callback = this[messageCallbacksSymbol].get(parsed['id']);
+      if (callback)
+        callback(parsed['error'] || null, parsed['result'] || null);
+      this[messageCallbacksSymbol].delete(parsed['id']);
+    } else {
+      this.emit(parsed['method'], parsed);
```

Comment:
Does the `method` field include the domain name too? For example, "Runtime.executionContextCreated" rather than just "executionContextCreated". Seems the domain name may be important so listeners can subscribe to exactly the events they want.